### DISCUSSION TITLE
Is there a reason that the JSON response content-type does not indicate charset=utf-8?

### DISCUSSION THREAD

**Discussion by benwilber** [tokio-rs/axum#2424] [General]
Looking at this code here:

https://github.com/tokio-rs/axum/blob/main/axum/src/json.rs#L194

It's just doing `application/json`

[But the JSON spec requires UTF-8 encoding except for "closed systems".](https://www.rfc-editor.org/rfc/rfc8259)

> JSON text exchanged between systems that are not part of a closed
   ecosystem MUST be encoded using UTF-8 [[RFC3629](https://www.rfc-editor.org/rfc/rfc3629)].

It seems to me that the most common case (not a closed system) will always require UTF-8 encoding for JSON responses.  I run into this frequently when using things like emojis in a JSON response and they show up as escape codes instead of emojis because Json(..) does not indicate the charset in the content-type.

**Comment by anarchodin** [comment]
From [RFC 8259, section 11](https://www.rfc-editor.org/rfc/rfc8259.html#section-11):

> Note:  No "charset" parameter is defined for this registration.
>    Adding one really has no effect on compliant recipients.

**Reply by davidpdrsn** [reply]
Ah I didn’t know this. But that possibly explains why the mime crate (which axum uses) doesn’t have a charset for application/json.
